src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+import { useEffectOnce, useMount } from 'react-use';
+import {
+  EuiButton,
+  EuiInMemoryTable,
+  EuiPageContent,
+  EuiPageContentBody,
+  EuiSpacer,
+  EuiSearchBarProps,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { FormattedMessage } from '@osd/i18n/react';
+import { getListBreadcrumbs } from '../breadcrumbs';
+import { PointInTimeManagementContext } from '../../types';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { getDataSources } from '../utils';
+import { EmptyState } from './empty_state';
+import { PageHeader } from './page_header';
+
+export interface DataSourceItem {
+  id: string;
+  title: string;
+  sort: string;
+}
+
+const PITTable = ({ history }: RouteComponentProps) => {
+  const {
+    setBreadcrumbs,
+    savedObjects,
+    notifications: { toasts },
+  } = useOpenSearchDashboards<PointInTimeManagementContext>().services;
+
+  useMount(() => {
+    setBreadcrumbs(getListBreadcrumbs());
+  });
+
+  // TODO: update this for cases when some data source name is default
+  const defaultDataSource: DataSourceItem = { id: '', title: 'Default', sort: '0' };
+
+  // TODO: use APIs to fetch PITs and update the table and message
+  const [pits] = useState([]);
+  const [message] = useState(<EmptyState />);

```

Comment:
Why is this a component - not sure if we should use state to save components, can we use a boolean to either show zero state or table instead ?

Reply:
So we will be adding and using `setMessage` to update the message based on the response from the get PIT API. There can be other states like error fetching data.